includes/admin/class-wc-stripe-admin-notices.php

```diff
@@ -240,6 +242,26 @@ public function payment_methods_check_environment() {
 				$this->add_admin_notice( $method, 'notice notice-error', sprintf( __( '%1$s is enabled - it requires store currency to be set to %2$s', 'woocommerce-gateway-stripe' ), $method, implode( ', ', $gateway->get_supported_currency() ) ), true );
 			}
 		}
+
+		if ( ! WC_Stripe_Feature_Flags::is_upe_enabled() || ! WC_Stripe::get_instance()->is_upe_enabled() ) {
+			return;
+		}
+
+		foreach ( WC_Stripe_UPE_Payment_Gateway::UPE_AVAILABLE_METHODS as $method_class ) {
+			if ( WC_Stripe_UPE_Payment_Method_CC::class === $method_class ) {
+				continue;
+			}
```

Comment:
I suppose not setting a value for this protected property on the CC method and simply skipping it should be fine. It looks like credit/debit cards is the only payment method without any currency restrictions. We will soon need an `array_filter` function like [this one](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L471-L484) in the payment gateway, so that we can filter payment methods at checkout by the selected currency as well, but we should be able to reuse the logic here. 

It looks like Alipay will require some more complicated logic here, as apparently the [required currency is dependent on the user's location](https://stripe.com/docs/sources/alipay#create-source) in a few cases, but we can cross that rickety bridge when we come to it.